pkg/manager/internal.go

```diff
@@ -346,23 +366,51 @@ func (cm *controllerManager) waitForCache() {
 	cm.started = true
 }
 
-func (cm *controllerManager) startLeaderElection() (err error) {
+func (cm *controllerManager) startLeaderElectionRunnable(leaderElectionID string) {
+	// Create resource lock
+	resourceLock, err := cm.newResourceLock(cm.config, cm.recorderProvider, crleaderelection.Options{
+		LeaderElection:          true,
+		LeaderElectionID:        leaderElectionID,
+		LeaderElectionNamespace: cm.leaderElectionNamespace,
+	})
+
+	// Controllers block, but we want to return an error if any have an error starting.
+	// Write any Start errors to a channel so we can return them
+	if err != nil {
+		cm.errChan <- err
+	}
+
+	err = cm.startLeaderElection(resourceLock, leaderelection.LeaderCallbacks{
+		OnStartedLeading: func(_ context.Context) {
+			cm.mu.Lock()
+			defer cm.mu.Unlock()
+
+			cm.electedLeaderElectionIDs[leaderElectionID] = true
+
+			runnables := cm.leaderElectionRunnables[leaderElectionID]
+			for _, r := range runnables {
+				runnable := r
+				go func() {
+					cm.errChan <- runnable.Start(cm.internalStop)
+				}()
+			}
+		},
+		OnStoppedLeading: func() {
+			cm.errChan <- fmt.Errorf("runnable leader election lost")
```

Comment:
this'll kill the entire controller.  Don't we want to continue operating?  What's the intended behavior here?  Do we need per-le-group stop channels?

Reply:
Yes, i think per-le-group stop channels would be reasonable.